langchain4j-oracle/src/main/java/dev/langchain4j/data/document/loader/oracle/OracleDocumentLoader.java

```
@@ -0,0 +1,170 @@
+            JsonNode rootNode = mapper.readTree(pref);
+            JsonNode fileNode = rootNode.path("file");
+            JsonNode dirNode = rootNode.path("dir");
+            JsonNode ownerNode = rootNode.path("owner");
+            JsonNode tableNode = rootNode.path("tablename");
+            JsonNode colNode = rootNode.path("colname");
+
+            if (fileNode.textValue() != null) {
+                String filename = fileNode.textValue();
+                Document doc = loadDocument(filename, pref);
+                if (doc != null) {
+                    documents.add(doc);
+                }
+            } else if (dirNode.textValue() != null) {
+                String dir = dirNode.textValue();
+                Path root = Paths.get(dir);
+                Files.walk(root).forEach(path -> {
+                    if (path.toFile().isFile()) {
+                        Document doc = loadDocument(path.toFile().toString(), pref);
+                        if (doc != null) {
+                            documents.add(doc);
+                        }
+                    }
+                });
+            } else if (colNode.textValue() != null) {
+                String column = colNode.textValue();
+
+                String table = tableNode.textValue();
+                String owner = ownerNode.textValue();
+                if (table == null) {
+                    throw new RuntimeException("Missing table in preference");
+                }
+                if (owner == null) {
+                    throw new RuntimeException("Missing owner in preference");
+                }
+
+                documents.addAll(loadDocuments(owner, table, column, pref));
+            } else {
+                throw new RuntimeException("Invalid preference");
+            }
+        } catch (IOException | RuntimeException ex) {
+            String message = ex.getCause() != null ? ex.getCause().getMessage() : ex.getMessage();
```

You don't need to catch the RuntimeException, and I think we should re-throw the exception if we can't recover from the IOException: https://www.baeldung.com/java-exceptions